.github/workflows/build.yml

```diff
@@ -49,11 +49,40 @@ jobs:
           ./gradlew -p wire-library build
           ./gradlew build
 
+  native:
+    runs-on: macOS-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Validate Gradle Wrapper
+        uses: gradle/wrapper-validation-action@v1
+
+      - name: Configure JDK
+        uses: actions/setup-java@v1
+        with:
+          java-version: 14
+
+      - name: Check test files
+        run: |
+          ./gradlew clean generateTests
+          if [ ! -z "$(git status --porcelain)" ]; then git diff; echo -e "\nTest files changed. Did you run ./gradlew generateTests?"; exit 1; fi
+
+      - name: Install Chrome
+        run: |
+          brew update
+          brew cask install google-chrome
+
+      - name: Test
+        run: |
+          ./gradlew -p wire-library build
```

Comment:
Can further improve this to only run JVM tests on Ubuntu to reduce the time spent on downloading dependencies for JS and Native, and run Native and JS tests on macOS only.

Reply:
Would be nice if this was built-in somehow, but I suspect we want to do this ourselves with properties.

Something like `./gradlew ... -Pkjs=false -Pknative=false` and then we do like
```groovy
if (getProperty("kjs") != "false")) {
  js {
    browser()
  }
}
```

That way everything is on-by-default but you can disable a platform if need-be.

Reply:
Hmm, I thought about approaching this by just invoking platform-specific tasks instead of `build`, I wonder though if there'll be something that runs during configuration that'll break the builds. I'm gonna try this first cause it won't require modifying any build files, if it fails I'll try introducing properties.

Reply:
My concern with executing platform-specific tasks is that it requires constant maintenance to opt-in rather than opt-out. With the system property approach, every build is run by default and if we want to optimize CI we disable targets. This ensures that we never forget a target and that as new modules come in that they're covered.

Reply:
That's a great point, definitely see the value of opt-out vs opt-in in this case. I'll introduce properties for JS and Native; not sure how to got about JVM, but ultimately it'd be nice to avoid running tests for JVM-only modules on macOS.